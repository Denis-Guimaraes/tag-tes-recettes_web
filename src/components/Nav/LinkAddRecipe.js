// NPM import
import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FaFileSignature } from 'react-icons/fa';

// Local import

// Styles
import './nav.scss';

// Code
export const LinkAddRecipe = ({ location }) => {
  const iconClass = classNames('nav-item__icon', { 'nav-item__icon--current': location.pathname === '/ajouter' });
  return (
    <li>
      <Link to="/ajouter" className="nav-item">
        <FaFileSignature className={iconClass} />Ajouter
      </Link>
    </li>
  );
};

LinkAddRecipe.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

// Export
export default withRouter(LinkAddRecipe);
