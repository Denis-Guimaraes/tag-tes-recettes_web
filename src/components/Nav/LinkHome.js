// NPM import
import React from 'react';
import { Link } from 'react-router-dom';
import { FaHome } from 'react-icons/fa';

// Local import

// Styles
import './nav.scss';

// Code
const LinkHome = () => (
  <li>
    <Link to="/" className="nav-item">
      <FaHome className="nav-item__icon" />Accueil
    </Link>
  </li>
);

// Export
export default LinkHome;
