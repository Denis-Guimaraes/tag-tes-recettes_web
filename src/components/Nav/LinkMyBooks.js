// NPM import
import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FaBook } from 'react-icons/fa';

// Local import

// Styles
import './nav.scss';

// Code
const LinkMybooks = ({ location }) => {
  const iconClass = classNames('nav-item__icon', { 'nav-item__icon--current': location.pathname === '/mes-livres' });
  return (
    <li>
      <Link to="/mes-livres" className="nav-item">
        <FaBook className={iconClass} />Mes livres
      </Link>
    </li>
  );
};

LinkMybooks.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

// Export
export default withRouter(LinkMybooks);
