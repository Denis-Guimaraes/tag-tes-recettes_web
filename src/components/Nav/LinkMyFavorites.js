// NPM import
import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FaFileAlt } from 'react-icons/fa';

// Local import

// Styles
import './nav.scss';

// Code
const LinkMyFavorites = ({ location }) => {
  const iconClass = classNames('nav-item__icon', { 'nav-item__icon--current': location.pathname === '/mes-recettes' });
  return (
    <li>
      <Link to="/mes-favorites" className="nav-item">
        <FaFileAlt className={iconClass} />Mes favorites
      </Link>
    </li>
  );
};

LinkMyFavorites.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

// Export
export default withRouter(LinkMyFavorites);
