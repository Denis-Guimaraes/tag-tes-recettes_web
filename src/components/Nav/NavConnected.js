// NPM import
import React from 'react';

// Local import
import LinkMyRecipes from './LinkMyRecipes';

// Styles
import './nav.scss';

// Code
const NavConnected = () => (
  <ul className="nav-list">
    <LinkMyRecipes />
  </ul>
);

// Export
export default NavConnected;
