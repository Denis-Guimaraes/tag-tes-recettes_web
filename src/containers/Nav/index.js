// NPM import
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// Local import
import Nav from 'src/components/Nav';

// Code
const mapStateToProps = state => ({
  userToken: state.user.userToken,
});

const mapDispatchToProps = () => ({});

// Export
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Nav));
