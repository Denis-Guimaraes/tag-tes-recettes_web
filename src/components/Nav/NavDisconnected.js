// NPM import
import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

// Local import
import LinkSignin from './LinkSignin';
import LinkSignup from './LinkSignup';
import LinkHome from './LinkHome';

// Styles
import './nav.scss';

// Code
export const NavDisconnected = ({ location }) => (
  <ul className="nav-list">
    {location.pathname !== '/connexion' && <LinkSignin />}
    {location.pathname !== '/' && <LinkHome />}
    {location.pathname !== '/inscription' && <LinkSignup />}
  </ul>
);

NavDisconnected.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

// Export
export default withRouter(NavDisconnected);
