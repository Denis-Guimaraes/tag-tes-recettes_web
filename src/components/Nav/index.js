// NPM import
import React from 'react';
import PropTypes from 'prop-types';

// Local import
import NavDisconnected from './NavDisconnected';
import NavConnected from './NavConnected';

// Styles
import './nav.scss';

// Code
const Nav = ({ userToken }) => (
  <nav className="nav">
    {userToken.length <= 0 ? <NavDisconnected /> : <NavConnected />}
  </nav>
);

Nav.propTypes = {
  userToken: PropTypes.string.isRequired,
};

// Export
export default Nav;
