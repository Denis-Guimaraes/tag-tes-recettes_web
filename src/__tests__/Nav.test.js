/* eslint-disable no-undef */
// NPM import
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// Local import
import Nav from '../components/Nav';
import NavDisconnected from '../components/Nav/NavDisconnected';
import NavConnected from '../components/Nav/NavConnected';

describe('Nav', () => {
  it('match snapshot', () => {
    const props = {
      userToken: '',
    };
    const wrapper = shallow(<Nav {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('contain component NavDisconnected', () => {
    const props = {
      userToken: '',
    };
    const wrapper = shallow(<Nav {...props} />);
    expect(wrapper.exists(NavConnected)).toBe(false);
    expect(wrapper.exists(NavDisconnected)).toBe(true);
  });
  it('contain component NavConnected', () => {
    const props = {
      userToken: '111',
    };
    const wrapper = shallow(<Nav {...props} />);
    expect(wrapper.exists(NavConnected)).toBe(true);
    expect(wrapper.exists(NavDisconnected)).toBe(false);
  });
});
