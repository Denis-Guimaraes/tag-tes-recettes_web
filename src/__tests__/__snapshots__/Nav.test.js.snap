// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav match snapshot 1`] = `
<nav
  className="nav"
>
  <withRouter(NavDisconnected) />
</nav>
`;
